src/vcpkg/build.cpp

```diff
@@ -1016,11 +1050,9 @@ namespace vcpkg::Build
         auto buildpath = paths.build_dir(action.spec);
         if (!fs.exists(buildpath, IgnoreErrors{}))
         {
-            std::error_code err;
-            fs.create_directory(buildpath, err);
-            Checks::check_exit(
-                VCPKG_LINE_INFO, !err.value(), "Failed to create directory '%s', code: %d", buildpath, err.value());
+            fs.create_directory(buildpath, VCPKG_LINE_INFO);
```

Comment:
As usual, this TOCTOU-esque pattern is unfortunate. It would be nice to have a simple `fs.ensure_directory(buildpath, VCPKG_LINE_INFO)` that does the Right Thing.

Reply:
I'm pretty sure `create_directory` already *is* that Right Thing (it isn't supposed to fail for existing directories) but doing that debugging seemed out of scope for this change.